src/main/java/TaskManager.java

```diff
@@ -0,0 +1,237 @@
+            throw new EvaException("Oh no! The task number you provided is out of range.\n" +
+                    "Please provide a valid task number between 1 and " + count + ".");
+        }
+
+        tasks.get(taskNumber).setMarkAsDone();
+
+        System.out.println("Great! This task is marked as done: ");
+        System.out.println(tasks.get(taskNumber).toString());
+        System.out.println("Well done! ;)");
+        System.out.println(HORIZONTAL_LINE);
+
+        saveTasksToFile();
+    }
+
+    public void unmarkTask(String line) throws EvaException {
+
+        int taskNumber = extractDigit(line) - 1;
+
+        if (taskNumber < 0 || taskNumber >= count) {
+            throw new EvaException("Oh no! The task number you provided is out of range.\n" +
+                    "Please provide a valid task number between 1 and " + count + ".");
+        }
+
+        tasks.get(taskNumber).setMarkAsNotDone();
+
+        System.out.println("Ok, This task is marked as not done yet: ");
+        System.out.println(tasks.get(taskNumber).toString());
+        System.out.println(HORIZONTAL_LINE);
+
+        saveTasksToFile();
+    }
+
+    public void deleteTask(String line) throws EvaException {
+        int taskNumber = extractDigit(line) - 1;
+
+        if (taskNumber < 0 || taskNumber >= count) {
+            throw new EvaException("Oh no! The task number you provided is out of range.\n" +
+                    "Please provide a valid task number between 1 and " + count + ".");
+        }
+
+        String temp = tasks.get(taskNumber).toString();
+
+        tasks.remove(taskNumber);
+        count = count - 1;
+
+        System.out.println("Okay. I have deleted task " + (taskNumber + 1) + ".");
+        System.out.println(temp);
+        printNumTasks(count - 1);
+        System.out.println(HORIZONTAL_LINE);
+
+        saveTasksToFile();
+    }
+
+    public static int extractDigit(String input) {
+        String numberString = input.replaceAll("[^0-9]", "");
+        return Integer.parseInt(numberString);
+    }
+
+    public void printTodo(String line) throws EvaException {
+
+        String todoDesc = line.replaceFirst("todo", "").trim();
+
+        if (todoDesc.isEmpty()) {
+            throw new EvaException("On no! The description of a todo cannot be empty." +
+                    " \nPlease try again by typing todo (name of task).");
+        }
+
+        tasks.add(new Todo(todoDesc));
+
+        System.out.println("Okay, I've added this todo: ");
+        System.out.println(tasks.get(count).toString());
+        printNumTasks(count);
+        System.out.println(HORIZONTAL_LINE);
+
+        count++;
+
+        saveTasksToFile();
+    }
+
+    public void printDeadline(String line) throws EvaException {
+
+        String[] parts = line.replaceFirst("deadline", "").split("/by");
+
+        if (parts.length < 2) {
+            throw new EvaException("Oh no! The deadline command must have a description and a by time." +
```

Comment:
for easy maintenance, better to declare a variable and use the variable in the exception throwing